### ISSUE TITLE
`declaringClass` of "enum-as-POJO" not removed for `ObjectMapper` with a naming strategy

### ISSUE THREAD

**Issue by Tynakuh** [FasterXML/jackson-databind#2365]
Happens in 2.9.9.

When you change ObjectMapper to use a different naming strategy, i.e:
```
ObjectMapper mapper = new ObjectMapper();
mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
```

When serialising/using writeToString(Object value) for an object such as:
```
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum Foo
{
    BAR("Foo", 0),
...
```

"declaring_class" will be added to the JSON payload.

Normally this is removed at:
`com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java:1139 `but because in
`com.fasterxml.jackson.databind.introspect.BasicBeanDescription#removeProperty` (It's only usage as far as I can see) we aren't using InternalName, declaringClass is not removed. Potential fix here is to change prop.getName() to prop.getInternalName(), a quick test of that seemed to work in my case.

**Comment by LukasKarasek** [comment]
Hello, I have a question, 

I had in the project older Jackson libraries 2.9.9 and I got this error with declaring class when serialize my enum. 
So I update all Jackson lib to 2.10.1 and I have a custom config for Jackson 

`jackson:
    default-property-inclusion: NON_EMPTY
    property-naming-strategy: SNAKE_CASE
    mapper-feature: ACCEPT_CASE_INSENSITIVE_PROPERTIES
    time-zone: 'UTC'
    serialization:
      write-dates-as-timestamps: true
      write-date-timestamps-as-nanoseconds: false
      write-bigdecimal-as-plain: true`

JSON response :
{
         "day_count_methods": [
        {
            "description": "Actual/365 Fixed, Actual/365",
            "name": "DCM_ACTUAL_365",
            "declaring_class": "com.f...."
        },
        {
            "description": "Actual/360, Act/360, A/360",
            "name": "DCM_ACTUAL_360",
            "declaring_class": "com.f..."
        },

Any idea how to fix this ? , Thanks

**Comment by cowtowncoder** [comment]
Interesting. Thank you for reporting this -- sounds like a bug indeed! I'll verify that suggested fix works: I suspect it should in some form: does for this property, but changing public method to change lookup might break other handling. But either addition of secondary lookup or (if need be) adding new method should work.

**Comment by cowtowncoder** [comment]
Interestingly enough, I can reproduce this on 2.10 branch (including 2.10.2) but NOT on 2.11.
So I think something got fixed since 2.10, although none of issues fixed so far seems related.
Still, adding test to verify fix I consider this resolved for 2.11 (and 3.0 via master); will add further notes if I find something explaining which change resulting in fix,.

**Comment by LukasKarasek** [comment]
Thanks 
I delete my post because first I thought it was fixed in 2.10.1 but then I realize that this is still ongoing ticket with this bug. 
I did a workaround on our side and will wait for fixed version. 
L.

**Comment by cowtowncoder** [comment]
Ok. At this point I think I will not try fixing it for 2.10.x partly since it is resolved for 2.11, and partly since I suspect there is non-trivial risk for regression for 2.10.x. And mostly I guess as I really hope to get 2.11.0 released relatively soon (early Feb 2019).

**Comment by LukasKarasek** [comment]
OK, thanks , I'll try when it will release .
